x/cdp/keeper/draw.go

```diff
@@ -175,8 +175,11 @@ func (k Keeper) ValidatePaymentCoins(ctx sdk.Context, cdp types.CDP, payment sdk
 		}
 		return types.ErrInvalidPaymentDenom(k.codespace, cdp.ID, principalDenoms, paymentDenoms)
 	}
+	if payment.IsAnyGT(debt) {
+		return types.ErrPaymentExceedsDebt(k.codespace, payment, debt)
+	}
 	for _, dc := range payment {
-		dp, _ := k.GetDebt(ctx, dc.Denom)
+		dp, _ := k.GetDebtParam(ctx, dc.Denom)
 		proposedBalance := cdp.Principal.AmountOf(dc.Denom).Sub(dc.Amount)
 		if proposedBalance.GT(sdk.ZeroInt()) && proposedBalance.LT(dp.DebtFloor) {
```

Comment:
This doesn't take into account the fees, so it looks like someone could choose an amount to repay that leaves the remainder under the debt floor.

Reply:
Actually since fees increase every block, it will be hard to specify the exact amount to repay to hit zero principle.
Maybe the cli could calulate the amount for the users? Or they specify a max-repayment which repays untill all debt is cancelled?